lib/sshkit/utils.ex

```diff
@@ -2,4 +2,11 @@ defmodule SSHKit.Utils do
   def shellescape(value), do: value
 
   def shellquote(value), do: value
+
+  def strings_to_charlists(opts) when is_list(opts) do
+    Enum.map(opts, fn {k, v} -> {k, charlist(v)} end)    
```

Comment:
I've had a little time to think about this and - based off of your previous work - I think I've found a solution that's not too ugly.

Here's the PR: #22. I would be happy if you could take a look and let me know what you think.

It also covers the cases, where binaries are "hidden" in nested lists/keywords. I came to this solution by basically rephrasing the problem statement a little. You're right that we can't easily distinguish lists from char lists with guard clauses. However we can descend into all other structures that may contain a binary and recursively convert any binaries we find still…